src/lxc/log.c

```diff
@@ -43,23 +44,78 @@
 #define LXC_LOG_DATEFOMAT_SIZE  15
 
 int lxc_log_fd = -1;
+static int syslog_enable = 0;
 int lxc_quiet_specified;
 int lxc_log_use_global_fd;
 static int lxc_loglevel_specified;
 
 static char log_prefix[LXC_LOG_PREFIX_SIZE] = "lxc";
 static char *log_fname = NULL;
+static char *log_vmname = NULL;
 
 lxc_log_define(lxc_log, lxc);
 
+static int lxc_log_priority_to_syslog(int priority)
+{
+	switch (priority) {
+	case LXC_LOG_PRIORITY_FATAL:
+		return LOG_EMERG;
+	case LXC_LOG_PRIORITY_ALERT:
+		return LOG_ALERT;
+	case LXC_LOG_PRIORITY_CRIT:
+		return LOG_CRIT;
+	case LXC_LOG_PRIORITY_ERROR:
+		return LOG_ERR;
+	case LXC_LOG_PRIORITY_WARN:
+		return LOG_WARNING;
+	case LXC_LOG_PRIORITY_NOTICE:
+	case LXC_LOG_PRIORITY_NOTSET:
+		return LOG_NOTICE;
+	case LXC_LOG_PRIORITY_INFO:
+		return LOG_INFO;
+	case LXC_LOG_PRIORITY_TRACE:
+	case LXC_LOG_PRIORITY_DEBUG:
+		return LOG_DEBUG;
+	}
+
+	/* Not reached */
+	return LOG_NOTICE;
+}
+
+/*---------------------------------------------------------------------------*/
+static int log_append_syslog(const struct lxc_log_appender *appender,
+			     struct lxc_log_event *event)
+{
+	char *msg;
+	int rc;
+
+	if (!syslog_enable)
+		return 0;
+
+	rc = vasprintf(&msg, event->fmt, *event->vap);
```

Comment:
Please avoid using any GNU extensions. We try to be POSIX-compatible and although we've used non-GNU extensions before we try to avoid adding new ones. (We also build on Android so one should also try to keep in mind that even some standard POSIX features might be not available. Though I guess, `vasprintf()` is.  :)